wrangler-api/src/main/java/io/cdap/wrangler/api/DirectiveWithMetrics.java

```diff
@@ -0,0 +1,26 @@
+/*
+ * Copyright © 2023 Cask Data, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+package io.cdap.wrangler.api;
+
+import java.util.List;
+
+/**
+ * A {@link Directive} that can emit metrics
+ */
+public interface DirectiveWithMetrics extends Directive {
```

Comment:
Should this be implementing Directive? Can this not be another interface that Directive impl can implement when custom metrics are needed to be defined? 
interface DirectiveMetricDef: getServiceName, getAppEntityType, getAppEntityTypeValue
class DefaultDirectiveMetricDef impl DirectiveMetricDef with a constructor that takes in directive name with hardcoded service name and aet? 
ColumnExpression can implement DirectiveMetricDef with custom values. 
What do you think about this approach? 
(Redesign this as list of metric where needed)